core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -387,6 +416,12 @@ void validate(FailureCollector collector) {
             .withConfigProperty(PATH).withConfigProperty(PATTERN);
         }
       }
+
+      if (!containsMacro(ENABLE_EXTERNAL_ENTITIES) && Strings.isNullOrEmpty(enableExternalEntities)) {
```

Comment:
shouldn't have this check, as it should be allowed to not specify a value here for backwards compatibility.